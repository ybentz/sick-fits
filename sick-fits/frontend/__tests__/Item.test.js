import ItemComponent from '../components/Item'
import { shallow } from 'enzyme'

const fakeItem = {
  title: 'Bla',
  id: 'ID1',
  price: 5000,
  description: 'Testing is real cool',
  image: 'image.png',
  largeImage: 'large-image.png',
}

describe('<Item/>', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(<ItemComponent item={fakeItem} />)
  })

  it('renders and title and price properly', () => {
    const priceTag = wrapper.find('PriceTag')
    expect(priceTag.children().text()).toBe('$50')
    expect(wrapper.find('Title a').text()).toBe(fakeItem.title)
  })

  it('renders and title and price properly', () => {
    const img = wrapper.find('img')
    expect(img.prop('src')).toBe(fakeItem.image)
    expect(img.prop('alt')).toBe(fakeItem.title)
  })

  it('renders the buttons properly', () => {
    const buttons = wrapper.find('.buttonList')
    expect(buttons.children()).toHaveLength(3)
    expect(buttons.find('Link').exists()).toBe(true)
    expect(buttons.find('AddToCart').exists()).toBe(true)
    expect(buttons.find('DeleteItem').exists()).toBe(true)
  })
})
