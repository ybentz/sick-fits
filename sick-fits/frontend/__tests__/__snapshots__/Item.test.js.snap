// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders properly 1`] = `
<DocumentFragment>
  <div
    class="ItemStyles__Item-sc-16pk14u-0 hZbgff"
  >
    <img
      alt="Bla"
      src="image.png"
    />
    <h3
      class="Title-sc-16nq74k-0 cBlaqK"
    >
      <a
        href="/item?id=ID1"
      >
        Bla
      </a>
    </h3>
    <span
      class="PriceTag-nwbk6t-0 hBhTMo"
    >
      $50
    </span>
    <p>
      Testing is real cool
    </p>
    <div
      class="buttonList"
    >
      <a
        href="/update?id=ID1"
      >
        Edit ✏️
      </a>
      <button>
        Add to Cart 🛒
      </button>
      <button>
        Delete Item
      </button>
    </div>
  </div>
</DocumentFragment>
`;
